tapchannel/aux_invoice_manager.go

```diff
@@ -197,6 +198,15 @@ func (s *AuxInvoiceManager) handleInvoiceAccept(_ context.Context,
 	// Convert the total asset amount to milli-satoshis using the price from
 	// the accepted quote.
 	rfqID := htlc.RfqID.ValOpt().UnsafeFromSome()
+
+	assetID := s.assetIDFromQuote(rfqID)
+	for _, v := range htlc.Balances() {
+		if v.AssetID.Val != *assetID {
```

Comment:
To avoid dereference of nil at `*assetID` I think `assetIDFromQuote` should return `asset.ID, error` (error in case of missing policy for HTLC).

Reply:
This is unaddressed. By returning the identifier directly, we can do `if !ident.HasId() {`.